build-artifacts/project-template-gradle/src/main/java/com/tns/RuntimeHelper.java

```diff
@@ -178,22 +182,43 @@ public static Runtime initRuntime(Application app) {
                     }
                 }
 
-                // runtime needs to be initialized before the NativeScriptSyncService is enabled because it uses runtime.runScript(...)
-                if (NativeScriptSyncService.isSyncEnabled(app)) {
-                    NativeScriptSyncService syncService = new NativeScriptSyncService(runtime, logger, app);
-
-                    syncService.sync();
-                    syncService.startServer();
-
-                    // preserve this instance as strong reference
-                    // do not preserve in NativeScriptApplication field inorder to
-                    // make the code more portable
-                    // @@@
-                    // Runtime.getOrCreateJavaObjectID(syncService);
-                } else {
-                    if (logger.isEnabled()) {
-                        logger.write("NativeScript LiveSync is not enabled.");
+                try {
+                    java.lang.Class NativeScriptSyncService = Class.forName("com.tns.NativeScriptSyncService");
+                    Method isSyncEnabledMethod = NativeScriptSyncService.getMethod("isSyncEnabled", Context.class);
+                    Boolean isSyncEnabled = (Boolean)isSyncEnabledMethod.invoke(NativeScriptSyncService, app);
+
+                    // runtime needs to be initialized before the NativeScriptSyncService is enabled because it uses runtime.runScript(...)
+                    if (isSyncEnabled) {
+                        Constructor cons = NativeScriptSyncService.getConstructor(new Class[] {Runtime.class, Logger.class, Context.class});
```

Comment:
I think you could name this better. How about `Constructor syncServiceConstructor`?